music-rpc.ts

```diff
@@ -44,9 +44,13 @@ class AppleMusicDiscordRPC {
     } finally {
       // Ensure the connection is properly closed
       if (this.rpc.ipc) {
-        console.log("Closing connection to Discord RPC");
-        this.rpc.close();
-        this.rpc.ipc = undefined;
+        console.log("Attempting to close connection to Discord RPC");
+        try {
+          this.rpc.close();
+        } finally {
+          console.log("Connection to Discord RPC closed");
+          this.rpc.ipc = undefined;
+        }
       }
```

Comment:
Can you extract this block to something like `this.tryCloseRPC()` to avoid having too many nested statements?

Reply:
Makes sense, will do

Reply:
I have also put the `if` statement in that method, let me know if you want it to take it out again

Reply:
That's fine, LGTM